.github/workflows/build-and-deploy-production.yml

```diff
@@ -0,0 +1,133 @@
+  
+      - name: Copy app_offline and web config to publish folder
+        run: |
+          Copy-Item E:/web/Offline/app_offline.htm E:/web/${{env.SitePath}}-NEW -Recurse -Force;
+          Copy-Item E:/web/Offline/app_offline.htm E:/web/${{env.SitePath}} -Recurse -Force;
+          if (Test-Path -Path E:/web/${{env.SitePath}})
+          {
+            Remove-Item -Path 'E:/web/${{env.SitePath}}-NEW/web.config' -Force;
+            Copy-Item E:/web/${{env.SitePath}}/web.config E:/web/${{env.SitePath}}-NEW -Recurse -Force;
+          }
+          if (Test-Path -Path E:/web/${{env.SitePath}}-PREVIOUS){
+              Remove-Item -LiteralPath 'E:/web/${{env.SitePath}}-PREVIOUS' -Force -Recurse
+            }        
+
+      - name: Sleep for 5 seconds
+        run: Start-Sleep -s 5 
+
+      - name: Switch deployment and published folders restarting apppool/webapp if necessary
+        run: |
+
+          Import-Module WebAdministration; 
+          $currentRetry = 1;
+          $backupRetry = 1;
+          $success = $false;
+          $backupSuccess = $false;
+          do{
+            echo "Attempting folder rename $currentRetry"  
+            try {
+              Rename-Item -Path 'E:/web/${{env.SitePath}}' -NewName '${{env.SitePath}}-PREVIOUS' 
+              Rename-Item -Path 'E:/web/${{env.SitePath}}-NEW' -NewName '${{env.SitePath}}' 
+              $success = $true;
+            }
+            catch {
+              echo "Rename failed due to following Catch error:`n"
+              echo $PSItem.Exception.Message
+              echo "`n"
+              Start-Sleep -s 2
+              $currentRetry = $currentRetry + 1;
+            }
+            finally {
+              if ($currentRetry -ge 10) {
+                  echo "Rename keeps failing; restarting AppPool/Site as last resort`n"
+                  echo "Attempting to restart AppPool`n"
+                  do{
+                      $status = Get-WebAppPoolState -name '${{env.AppPoolName}}'
+                      if ($status.Value -eq "Stopped") {
+                          start-WebAppPool ${{env.AppPoolName}}
+                          echo "AppPool restarted`n---------`n"
+                          $backupSuccess = $true;
+                      }
+                      else {
+                          if ($backupRetry -ge 10) {
+                             throw "AppPool restart keeps failing."
+                          }
+                          echo "AppPool not stopped yet; Re-attempt #$backupRetry"
+                          Start-Sleep -s 10
+                          $backupRetry = $backupRetry + 1;
+                      }
+                  }
+                  while (!$backupSuccess -and $backupRetry -le 10)
+                  $backupRetry = 1;
+                  $backupSuccess = $false;
+                  echo "Attempting to restart Website`n"
+                  do{
+                      $status = Get-webtate -name '${{env.SiteName}}'
```

Comment:
Just a question, is "Get-webtate" a command or a typo here?

Reply:
good spot - fixed